### PR TITLE
Minor performance improvements

### PR DESCRIPTION
**Pull request by nmittler**
These were found when profiling benchmarks for #11293

### CODE REVIEW

File: pkg/spiffe/spiffe.go [hunk 1]
```diff
@@ -10,14 +10,19 @@ import (
...
 
-var trustDomain = defaultTrustDomain
+var (
+	trustDomain = defaultTrustDomain
+)
 
 func SetTrustDomain(value string) {
-	trustDomain = value
+	// Replace special characters in spiffe
+	trustDomain = strings.Replace(value, "@", ".", -1)
```
Comment (ozevren) [inline pkg/spiffe/spiffe.go]: There doesn't seem to be a test for this?

Comment (nmittler) [reply pkg/spiffe/spiffe.go]: updated the test to account for this

**Comment by nmittler** [comment]
@ozevren PTAL

**Comment by nmittler** [comment]
@incfly mind taking a look? This is the same PR you approved the other day, just against the master branch.

**Comment by incfly** [comment]
Thanks for doing at master again! But wouldn't the release-1.1 be merged back to master? Or we need to backport in two different branches ourselves?

**Comment by nmittler** [comment]
@incfly 

> Thanks for doing at master again! But wouldn't the release-1.1 be merged back to master? Or we need to backport in two different branches ourselves?

My understanding is that post-1.1 work should be done in master for now to allow us to continue to make progress. Yes, it's true that some 1.1 changes will get merged back to master, but the theory is that they should be relatively small patches at this point.  Larger changes will be going from master->1.1 going forward.

**Pull request merged**